### PR TITLE
Relations manager tests

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Completes #2107

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import { renderHook, act } from '@testing-library/react-hooks';
```
Comment (nicolad) [inline assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts]: `RelationsManager.addRelation.test.ts` looks a bit strange to me.
What do you say about creating `useRelationsManager` folder on the same level as `useRelationsManager.ts`
and moving this test there as a `addRelation.test.ts` file ?

Comment (manzoorwanijk) [reply assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts]: That's a good suggestion

Comment (nicolad) [reply assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts]: and `useRelationsManager.ts` might be renamed to `index.ts` in that folder.

File: assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		const { result } = renderHook(() => useRelationsManager(relationalData));
+
+		const options: RelationFunctionProps = {
+			entity: 'datetimes',
+			entityId: existingRelationalEntityId,
+			relation: 'tickets',
+		};
+
+		// before updating relations
+		const relatedIdsBeforeAdd = result.current.getRelations(options);
```
Comment (nicolad) [inline assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts]: :+1:

File: assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts (hunk 1 above)
Comment (manzoorwanijk) [reply assets/prototype/application/services/apollo/relations/test/RelationsManager.addRelation.test.ts]: Changed

File: assets/prototype/application/services/apollo/relations/test/useRelationsManager/dropRelations.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		let relatedIds = result.current.getRelations(options);
+
+		const pathToRelatedEntityIds = Object.values(options);
+		const passedRelatedIds = pathOr([], pathToRelatedEntityIds, relationalData);
+
+		// before update
+		expect(relatedIds).toEqual(passedRelatedIds);
+
+		act(() => {
+			result.current.dropRelations(pick(['entity', 'entityId'], options));
```
Comment (nicolad) [inline assets/prototype/application/services/apollo/relations/test/useRelationsManager/dropRelations.test.ts]: nice usage of ramdas function :+1:

**Pull request merged**